### PR TITLE
Remaining behat tests

### PR DESCRIPTION
**Pull request by bartoszpietrzak1994**
```
    Scenario: Having shipping method not filled with shipping information taken from previously placed order
        Given I browse my orders
        When I click reorder button next to the order "#00000666"
        And I proceed to the addressing step
        And I proceed to the shipping step
        Then "Golab Pocztowy" shipping method should not be selected

    Scenario: Having payment method not filled with payment information taken from previously placed order
        Given I browse my orders
        When I click reorder button next to the order "#00000666"
        And I proceed to the addressing step
        And I proceed to the shipping step
        And I proceed to the payment step
        Then "Paypal" payment method should not be selected
```

Scenarios fulfilled

Fixes https://github.com/Sylius/CustomerReorderPlugin/issues/23

### CODE REVIEW

File: features/reordering_previously_placed_order.feature [hunk 1]
```diff
@@ -10,15 +10,19 @@ Feature: Reordering previously placed order
         And there are 25 units of product "Angel T-Shirt" available in the inventory
         And this product is tracked by the inventory
         And the store ships everywhere for free
+        And the store also allows shipping with "Golab Pocztowy"
```
Comment (GSadee) [inline features/reordering_previously_placed_order.feature]: `Post Pigeon` 🎉 be consistent with english names 😄

File: tests/Behat/Context/Reorder/ReorderContext.php [hunk 2]
```diff
@@ -152,27 +163,19 @@ public function iShouldBeNotifiedThatNoneOfItemsFromPreviouslyPlacedOrderIsAvail
...
-
-    /**
-     * @Then I should not have the shipping method section copied from order :orderNumber
+     * @Then :shippingMethod shipping method should not be selected
      */
-    public function iShouldNotHaveTheShippingMethodSectionFilledWithInformationTakenFromOrder(string $orderNumber): void
+    public function shippingMethodShouldNotBeSelected(string $shippingMethod): void
     {
-
+        assert($this->selectShippingPage->isShippingMethodSelected($shippingMethod) === false);
```
Comment (GSadee) [inline tests/Behat/Context/Reorder/ReorderContext.php]: `Assert::false($this->selectShippingPage->isShippingMethodSelected($shippingMethod))`?

File: tests/Behat/Context/Reorder/ReorderContext.php [hunk 3]
```diff
@@ -188,15 +191,15 @@ public function iProceedToTheAddressingStep(): void
      */
     public function iProceedToTheShippingStep(): void
     {
-
+        $this->session->getPage()->find('css', '#next-step')->click();
```
Comment (GSadee) [inline tests/Behat/Context/Reorder/ReorderContext.php]: This behaviour should be in page.

And for example, [AddressPageInterface](https://github.com/Sylius/Sylius/blob/master/src/Sylius/Behat/Page/Shop/Checkout/AddressPageInterface.php#L93) has method `nextStep` which you can use here

File: tests/Behat/Page/SelectPaymentPageInterface.php [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+declare(strict_types=1);
+
+namespace Tests\Sylius\CustomerReorderPlugin\Behat\Page;
+
+use \Sylius\Behat\Page\Shop\Checkout\SelectPaymentPageInterface as BaseSelectPaymentPageInterface;
+
+interface SelectPaymentPageInterface extends BaseSelectPaymentPageInterface
+{
+    public function isPaymentMethodSelected(string $paymentMethod): bool;
```
Comment (GSadee) [inline tests/Behat/Page/SelectPaymentPageInterface.php]: `$paymentMethodName`?

File: tests/Behat/Context/Reorder/ReorderContext.php [hunk 5]
```diff
@@ -188,15 +198,15 @@ public function iProceedToTheAddressingStep(): void
...
+        $this->addresPage->nextStep();
     }
 
     /**
      * @When I proceed to the payment step
      */
     public function iProceedToThePaymentStep(): void
     {
-
+        $this->session->getPage()->find('css', '#next-step')->click();
```
Comment (GSadee) [inline tests/Behat/Context/Reorder/ReorderContext.php]: This behaviour should be in page as above

File: tests/Behat/Context/Reorder/ReorderContext.php [hunk 6]
```diff
@@ -124,23 +142,23 @@ public function iShouldBeNotifiedThatPromotionIsNoLongerEnabled(string $promotio
...
+        Assert::eq(count($this->session->getPage()->findAll('css', '.sylius-flash-message')), 0);
     }
 
     /**
      * @Then I should not proceed to my cart summary page
      */
     public function iShouldNotProceedToMyCartSummaryPage(): void
     {
-        assert(strpos($this->session->getCurrentUrl(), '/cart') === false);
+        Assert::false(strpos($this->session->getCurrentUrl(), '/cart'));
```
Comment (GSadee) [inline tests/Behat/Context/Reorder/ReorderContext.php]: You should check if the cart summary page is open on its page by using the method `isOpen`

File: tests/Behat/Context/Reorder/ReorderContext.php [hunk 7]
```diff
@@ -124,23 +142,23 @@ public function iShouldBeNotifiedThatPromotionIsNoLongerEnabled(string $promotio
...
+        Assert::eq(count($this->session->getPage()->findAll('css', '.sylius-flash-message')), $count);
     }
 
     /**
      * @Then I should not see any notifications
      */
     public function iShouldNotSeeAnyNotifications(): void
     {
-        assert(count($this->session->getPage()->findAll('css', '.sylius-flash-message')) === 0);
+        Assert::eq(count($this->session->getPage()->findAll('css', '.sylius-flash-message')), 0);
```
Comment (GSadee) [inline tests/Behat/Context/Reorder/ReorderContext.php]: That should be check on the page, not in the context

Comment (GSadee) [inline tests/Behat/Context/Reorder/ReorderContext.php]: And you can merge this method with the one above by setting the default value `int $count = 0`

File: features/reordering_previously_placed_order.feature [hunk 8]
```diff
@@ -10,15 +10,19 @@ Feature: Reordering previously placed order
         And there are 25 units of product "Angel T-Shirt" available in the inventory
         And this product is tracked by the inventory
         And the store ships everywhere for free
+        And the store also allows shipping with "Post Pigeon"
+        And the store also allows shipping with "Hedwig Owl"
```
Comment (Zales0123) [inline features/reordering_previously_placed_order.feature]: `Hedwig Owl` ❤️

File: tests/Behat/Context/Reorder/ReorderContext.php [hunk 9]
```diff
@@ -18,12 +21,27 @@
...
     private $reorderEligibilityConstraintMessageFormatter;
 
+    /** @var SelectShippingPageInterface */
+    private $selectShippingPage;
+
+    /** @var SelectPaymentPageInterface */
+    private $selectPaymentPage;
+
+    /** @var AddressPageInterface */
+    private $addresPage;
```
Comment (Zales0123) [inline tests/Behat/Context/Reorder/ReorderContext.php]: `addresPage` -> `addressPage`

File: tests/Behat/Context/Reorder/ReorderContext.php (hunk 7 above)
Comment (Zales0123) [reply tests/Behat/Context/Reorder/ReorderContext.php]: Yeah, we should have something like `countFlashMessages` in the page class 👌

File: tests/Behat/Page/SelectPaymentPage.php [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+
+use \Sylius\Behat\Page\Shop\Checkout\SelectPaymentPage as BaseSelectPaymentPage;
+
+final class SelectPaymentPage extends BaseSelectPaymentPage implements SelectPaymentPageInterface
+{
+    public function isPaymentMethodSelected(string $paymentMethodName): bool
+    {
+        return null !== $this
+                ->getElement('payment_method_option', ['%payment_method%' => $paymentMethodName])
+                ->getAttribute('checked')
```
Comment (Zales0123) [inline tests/Behat/Page/SelectPaymentPage.php]: Wrong formatting

File: tests/Behat/Page/SelectShippingPage.php [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+
+use Sylius\Behat\Page\Shop\Checkout\SelectShippingPage as BaseSelectShippingPage;
+
+final class SelectShippingPage extends BaseSelectShippingPage implements SelectShippingPageInterface
+{
+    public function isShippingMethodSelected(string $shippingMethodName): bool
+    {
+        return null !== $this
+                ->getElement('shipping_method_option', ['%shipping_method%' => $shippingMethodName])
+                ->getAttribute('checked')
```
Comment (Zales0123) [inline tests/Behat/Page/SelectShippingPage.php]: As well as here

File: tests/Behat/Page/Checkout/SelectShippingPage.php [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+namespace Tests\Sylius\CustomerReorderPlugin\Behat\Page\Checkout;
+
+use Sylius\Behat\Page\Shop\Checkout\SelectShippingPage as BaseSelectShippingPage;
+
+final class SelectShippingPage extends BaseSelectShippingPage implements SelectShippingPageInterface
+{
+    public function isShippingMethodSelected(string $shippingMethodName): bool
+    {
+        return null !== $this->getElement('shipping_method_option', ['%shipping_method%' => $shippingMethodName])
+                ->getAttribute('checked');
```
Comment (Zales0123) [inline tests/Behat/Page/Checkout/SelectShippingPage.php]: Same here

File: tests/Behat/Page/Order/IndexPage.php [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+
+final class IndexPage extends BaseIndexPage implements IndexPageInterface
+{
+    public function clickReorderButtonNextToTheOrder(string $orderNumber): void
+    {
+        $orderData = $this->getOrderData($orderNumber);
+
+        $reorderButton = $orderData->find('css', 'td:last-child')->find('css', 'button');
+
+        if (null === $reorderButton || $reorderButton->getText() != 'Reorder') {
```
Comment (Zales0123) [inline tests/Behat/Page/Order/IndexPage.php]: Be consistent. If we have `null === $reorderButton` it should be also `'Reorder' !== $reorderButton->getText()`

File: tests/Behat/Page/Cart/SummaryPageInterface.php [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace Tests\Sylius\CustomerReorderPlugin\Behat\Page\Cart;
+
+use \Sylius\Behat\Page\Shop\Cart\SummaryPageInterface as BaseSummaryPageInterface;
+
+interface SummaryPageInterface extends BaseSummaryPageInterface
+{
+    public function checkout(): void;
+    public function countFlashMessages(): int;
+    public function assertFlashMessageWithTextExists(string $text): bool;
```
Comment (Zales0123) [inline tests/Behat/Page/Cart/SummaryPageInterface.php]: If the method is named `assert...` I would expect it to not return anything, but throw an exception if assertion failed. Or we could name it `doesFlashMessageWithTextExist` 😄

File: tests/Behat/Page/Checkout/SelectPaymentPage.php [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+namespace Tests\Sylius\CustomerReorderPlugin\Behat\Page\Checkout;
+
+use \Sylius\Behat\Page\Shop\Checkout\SelectPaymentPage as BaseSelectPaymentPage;
+
+final class SelectPaymentPage extends BaseSelectPaymentPage implements SelectPaymentPageInterface
+{
+    public function isPaymentMethodSelected(string $paymentMethodName): bool
+    {
+        return null !== $this->getElement('payment_method_option', ['%payment_method%' => $paymentMethodName])
+                ->getAttribute('checked');
```
Comment (Zales0123) [inline tests/Behat/Page/Checkout/SelectPaymentPage.php]: Why :( Can't it be:
```php
return null !== $this
    ->getElement('payment_method_option', ['%payment_method%' => $paymentMethodName])
    ->getAttribute('checked')
;
```

File: tests/Behat/Page/Order/IndexPage.php [hunk 16]
```diff
@@ -0,0 +1,44 @@
...
+        $reorderButton->click();
+    }
+
+    public function isReorderButtonVisibleNextToTheOrder(string $orderNumber): bool
+    {
+        $orderData = $this->getOrderData($orderNumber);
+
+        $actionButtonsText = $orderData->find('css', 'td:last-child')->getText();
+
+        return strpos($actionButtonsText, 'Reorder') !== false;
```
Comment (Zales0123) [inline tests/Behat/Page/Order/IndexPage.php]: If we use yoda condition, use it everywhere `return false !== strpos($actionButtonsText, 'Reorder');`

**Comment by Zales0123** [comment]
@bartoszpietrzak1994 could you please squash these 3 "Review fixes" commits? 😄

File: tests/Behat/Page/Order/IndexPage.php [hunk 17]
```diff
@@ -0,0 +1,44 @@
...
+use Behat\Mink\Element\NodeElement;
+use \Sylius\Behat\Page\Shop\Account\Order\IndexPage as BaseIndexPage;
+
+final class IndexPage extends BaseIndexPage implements IndexPageInterface
+{
+    public function clickReorderButtonNextToTheOrder(string $orderNumber): void
+    {
+        $orderData = $this->getOrderData($orderNumber);
+
+        $reorderButton = $orderData->find('css', 'td:last-child')->find('css', 'button');
```
Comment (pamil) [inline tests/Behat/Page/Order/IndexPage.php]: This reverts the solution I've implemented in #30 that prevents Behat failing when run together with Sylius/CustomerOrderCancellationPlugin.

File: tests/Behat/Page/Order/IndexPage.php [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+        }
+
+        $reorderButton->click();
+    }
+
+    public function isReorderButtonVisibleNextToTheOrder(string $orderNumber): bool
+    {
+        $orderData = $this->getOrderData($orderNumber);
+
+        $actionButtonsText = $orderData->find('css', 'td:last-child')->getText();
```
Comment (pamil) [inline tests/Behat/Page/Order/IndexPage.php]: This reverts the solution I've implemented in #30 that prevents Behat failing when run together with Sylius/CustomerOrderCancellationPlugin.

File: tests/Behat/Page/Order/IndexPage.php (hunk 17 above)
Comment (bartoszpietrzak1994) [reply tests/Behat/Page/Order/IndexPage.php]: Sorry! Fixed.

File: tests/Behat/Page/Cart/SummaryPage.php [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        $this->getSession()->getPage()->clickLink('Checkout');
+    }
+
+    public function countFlashMessages(): int
+    {
+        return count($this->getSession()->getPage()->findAll('css', '.sylius-flash-message'));
+    }
+
+    public function doesFlashMessageWithTextExists(string $text): bool
```
Comment (Zales0123) [inline tests/Behat/Page/Cart/SummaryPage.php]: `Exists` -> `Exist` :)

File: tests/Behat/Page/Order/IndexPage.php [hunk 20]
```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Sylius\CustomerReorderPlugin\Behat\Page\Order;
+
+use Behat\Mink\Element\NodeElement;
+use \Sylius\Behat\Page\Shop\Account\Order\IndexPage as BaseIndexPage;
```
Comment (Zales0123) [inline tests/Behat/Page/Order/IndexPage.php]: `\Sylius` -> `Sylius`

**Comment by Zales0123** [comment]
Thank you Bartosz! 🎉

**Pull request merged**